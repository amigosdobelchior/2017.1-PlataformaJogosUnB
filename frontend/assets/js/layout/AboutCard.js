import React from "react";
import {Card, Grid, Image} from "semantic-ui-react";

export default class AboutCardComponent extends React.Component {

    render () {

        return (
            <Card>
                <Card.Content>
                    <Card.Header>
                        {this.props.title}
                    </Card.Header>
                </Card.Content>
                <Card.Content>
                    <Card.Description>
                        <Grid centered columns={this.props.sizeGrid} divided>
                            <Grid.Row>
                                <Grid.Column>
                                    {this.props.description}
                                </Grid.Column>
                                <Grid.Column>
                                    <Image centered size="medium" src={this.props.image} />
                                </Grid.Column>
                            </Grid.Row>
                        </Grid>
                    </Card.Description>
                </Card.Content>
            </Card>
        );

    }
}
