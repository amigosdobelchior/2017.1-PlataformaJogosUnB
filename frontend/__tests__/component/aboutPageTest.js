import React from 'react';
import AboutCard from '../../assets/js/layout/AboutCard';
import {Grid, Card, Container} from "semantic-ui-react";
import renderer from 'react-test-renderer';

test('Test render AboutPage', () => {
  const component = renderer.create(
      <Grid padded="vertically">
          <Grid.Row color="blue">
              <Container>
                  <Card.Group itemsPerRow={1}>
                      <AboutCard
                          title="Titulo"
                          description="Descricao"
                          sizeGrid={1}
                      />
                  </Card.Group>
              </Container>
          </Grid.Row>
          <Grid.Row>
              <Container>
                  <Card.Group itemsPerRow={2}>
                      <AboutCard
                          title={"Venha conhecer"}
                          description={"desfrute de inumeros jogos"}
                          sizeGrid={1}
                      />
                      <AboutCard />
                  </Card.Group>
              </Container>
          </Grid.Row>
      </Grid>
	);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
