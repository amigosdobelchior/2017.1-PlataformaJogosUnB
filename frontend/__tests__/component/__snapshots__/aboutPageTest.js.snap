// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render AboutPage 1`] = `
<div
  className="ui vertically padded grid"
>
  <div
    className="blue row"
  >
    <div
      className="ui container"
    >
      <div
        className="ui one cards"
      >
        <div
          className="ui card"
          href={undefined}
          onClick={[Function]}
        >
          <div
            className="content"
          >
            <div
              className="header"
            >
              Titulo
            </div>
          </div>
          <div
            className="content"
          >
            <div
              className="description"
            >
              <div
                className="ui centered divided one column grid"
              >
                <div
                  className="row"
                >
                  <div
                    className="column"
                  >
                    Descricao
                  </div>
                  <div
                    className="column"
                  >
                    <img
                      alt={undefined}
                      className="ui medium centered image"
                      height={undefined}
                      src={undefined}
                      width={undefined}
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="ui container"
    >
      <div
        className="ui two cards"
      >
        <div
          className="ui card"
          href={undefined}
          onClick={[Function]}
        >
          <div
            className="content"
          >
            <div
              className="header"
            >
              Venha conhecer
            </div>
          </div>
          <div
            className="content"
          >
            <div
              className="description"
            >
              <div
                className="ui centered divided one column grid"
              >
                <div
                  className="row"
                >
                  <div
                    className="column"
                  >
                    desfrute de inumeros jogos
                  </div>
                  <div
                    className="column"
                  >
                    <img
                      alt={undefined}
                      className="ui medium centered image"
                      height={undefined}
                      src={undefined}
                      width={undefined}
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ui card"
          href={undefined}
          onClick={[Function]}
        >
          <div
            className="content"
          >
            <div
              className="header"
            />
          </div>
          <div
            className="content"
          >
            <div
              className="description"
            >
              <div
                className="ui centered divided grid"
              >
                <div
                  className="row"
                >
                  <div
                    className="column"
                  />
                  <div
                    className="column"
                  >
                    <img
                      alt={undefined}
                      className="ui medium centered image"
                      height={undefined}
                      src={undefined}
                      width={undefined}
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
